templates/shared_services/nexus-cert/terraform/appgateway.tf

```diff
@@ -0,0 +1,163 @@
+resource "azurerm_public_ip" "appgwpip" {
+  name                = "pip-nexus-${var.tre_id}"
+  resource_group_name = data.azurerm_resource_group.rg.name
+  location            = data.azurerm_resource_group.rg.location
+  allocation_method   = "Static"
+  sku                 = "Standard"
+  domain_name_label   = "nexus-${var.tre_id}"
+
+  lifecycle { ignore_changes = [tags] }
+}
+
+resource "azurerm_user_assigned_identity" "agw_id" {
+  resource_group_name = data.azurerm_resource_group.rg.name
+  location            = data.azurerm_resource_group.rg.location
+  name                = "id-agw-nexuscert-${var.tre_id}"
+
+  lifecycle { ignore_changes = [tags] }
+}
+
+resource "azurerm_application_gateway" "agw" {
```

Comment:
I think a better approach for this would be as part of executing the logic currently in make nexus-letsencrypt, this is done as a custom action on the bundle, which takes care of starting and stopping the service while it generates the cert. That way its only up and running while generating a cert and put into stand-by afterwards

Reply:
ok, so I assume this is also something you're working on...